### DISCUSSION TITLE
Confirming that RTL 8152/8153 driver works with DS920+ 7.2-64570 Update 1

### DISCUSSION THREAD

**Discussion by hero2017** [bb-qq/r8152#331] [Show and tell]
Keep in mind that even though I followed the instructions for installing this driver it still failed. That includes running the suggested command and even after rebooting. I had to uninstall the failed driver, reinstall it from Package center again and run the suggested command once again to make it work. Minor issue to keep in mind.

I ordered the Sabrent NT-S25G so hopefully it will run as expected even though it's not listed.

**Comment by hero2017** [comment]
I also ordered and tested Plugable ‎USBC-E2500 so here are some performance results for you:

Tested with iperf3 running on my DS920+ (with 20 GB ram) connected to the back USB port and my Dell G15 laptop with 2.5 Gbps ethernet port. Using Sodola 2.5 Gbps switch. Great results with the latest DSM which as of today is DSM 7.2-64570 Update 1 and latest RTL8152/8153 driver v2.17.1-1.

iperf3 test results (almost identical when comparing the Plugable 2.5 Gbps with the Sabrent 2.5 Gbps. No issues at all so far and overall very happy with the performance. I'm returning the Sabrent as it was $10 more and I only bought it for comparison. Btw the Sabrent is quite a bit bigger than the Plugable ethernet adapter. Both run cool to the touch as I feel almost no heat from them.

`
![image](https://github.com/bb-qq/r8152/assets/20324387/a0df717a-aac0-4e76-bb56-5a61f54dd926)

`

**Reply by genomez** [reply]
Have you tried it with DSM 7.2-64570 **Update 3**? I have been trying to install the Plugable and just can't get the driver to install with the extra command, removing the driver and multiple reboots. My DSM Info Center shows "USB 10/100/1G/2.5G LAN - Realtek Semiconductor Corp.", but I can't the Package Center error to ever clear. I'm also using a DS920+.

**Reply by ylcet** [reply]
No, only the DSM version I listed which is Update 1. What error do you get upon installation or after installation? Have you tried to uninstall it with/without the USB device plugged in? Lastly, what driver did you install exactly? Try the exact same version & file I installed. If all fails check the logs.

I had a problem at first too but after installing and it failing I rebooted a 2nd time before the 2nd installation attempt worked.

Note that someone else reported the same problem although with the same version of DSM as I used: https://github.com/bb-qq/r8152/issues/336

And another replied saying they were able to resolve it by upgrading their DSM from DSM 7.2-64570 to DSM 7.2.1-69057.

**Reply by genomez** [reply]
Thanks for the reply. I get a message that says either failed to repair or failed to install. 

I have tried it with and without the USB device plugged in and on the front and back USB ports. 

I have tried r8152-geminilake-2.17.1-1_7.2, but have also tried r8152-geminilake-2.16.3-4_7.2.

I'll check the DSM upgrade option. Thank you!

**Reply by Sjekke** [reply]
Are you running iperf in Docker? I run it on DS916+ and performance is not great. I use v1 and v2 of the Plugable hardware

**Comment by ylcet** [comment]
No I don't use docker. Not sure why performance would be worse on DS916+ but it could be that DS920+ has better hardware...faster cpu, ram. Could be the DSM version too I suppose.

**Reply by Sjekke** [reply]
So you run iperf native?

**Comment by genomez** [comment]
Here's an update. After following your links above I found someone who tried to stop and start the script and it finally started. However, in Package Manager it still says that there's an error and the package needs to be reinstalled, but the Plugable NIC is functioning properly. In fact, for the last 24 hours or so I've had no ethernet cables plugged into either of the onboard NIC's.
![image](https://github.com/bb-qq/r8152/assets/115205222/51791255-c1a0-4c90-b7be-1e3dbd1479ec)


Also, as a note for the person above, I am running iPerf3 in Docker on my DS920+ and these were the results.
![image](https://github.com/bb-qq/r8152/assets/115205222/dbef60e4-ef8d-4610-b967-bb90aae65142)

**Comment by TheS1R** [comment]
Why not run iperf3 natively on the NAS as opposed to in docker? It is included in the SynoCli Monitor Tools community-based package by[SynoCOmmunity](https://synocommunity.com/packages).

**Reply by Sjekke** [reply]
Good tip, I didn't even know it existed... installed it, opened terminal session and started it. Works good but how can i leave it always active. In docker iperf3 runs for me 24h/24h?

**Comment by TheS1R** [comment]
Launch a background process in a command window or script with output going to a file?
